src/devices_models/devices/common/add_dual_variable.jl

```diff
@@ -5,8 +5,8 @@ function add_dual_variable!(
 ) where {T <: PSY.Component, D <: AbstractDeviceFormulation}
     if !isempty(get_duals(model))
         devices = get_available_components(T, sys)
-        for constriant_key in get_duals(model)
-            assign_dual_variable!(container, constriant_key, devices, D)
+        for constraint_type in get_duals(model)
```

Comment:
The duals are keyed with a ConstraintKey`, are you sure you want to keep this name?

Reply:
This should actually fail. It should be
```
        for constraint_key in keys(get_duals(model))
            assign_dual_variable!(container, get_entry_type(constraint_key), devices, D)
        end
```
Same with other cases below.